src/simulations/simulator.py

```diff
@@ -134,12 +140,20 @@ def run_all_and_aggregate(self):
 
     return df, df_agg
 
+  def _extend_histogram(self, histogram, max_freq):
+    """Extends (or truncates) a frequency histogram to max_freq frequencies."""
+    if len(histogram) <= max_freq:
+      return histogram + [0] * (max_freq - len(histogram))
+    else:
+      return histogram[:max_freq] 
```

Comment:
Here the code truncates at freq = max_freq. Does it make sense that we merge the freq categories into one when freq >= max_freq? For example, when max_freq is set to 3, and the estimator returns freq = 1, 2, 3, 4, then we merge 3 and 4 into 3+. In this way, we won't miss a certain proportion of IDs so that two different histograms can be compared with each other in terms of their distributions.

Reply:
Lets say the seq of id's is 1,1,1,1,1,2,2,2,2,3,3,3,4,4,5.  So the full frequency histogram is then 15,10,6,3,1.  If we truncate this at max_freq=3, it becomes 15,10,6.  None of the id's are dropped, because the the 3+ bucket counts id's have a frequency of 3 or higher.  Let me know if this makes sense.